# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet

In [1]:
import pandas as pd
import numpy as np
from scipy import stats

In [2]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [4]:
data.head(5)

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

In [5]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4870 entries, 0 to 4869
Data columns (total 65 columns):
id                    4870 non-null object
ad                    4870 non-null object
education             4870 non-null int8
ofjobs                4870 non-null int8
yearsexp              4870 non-null int8
honors                4870 non-null int8
volunteer             4870 non-null int8
military              4870 non-null int8
empholes              4870 non-null int8
occupspecific         4870 non-null int16
occupbroad            4870 non-null int8
workinschool          4870 non-null int8
email                 4870 non-null int8
computerskills        4870 non-null int8
specialskills         4870 non-null int8
firstname             4870 non-null object
sex                   4870 non-null object
race                  4870 non-null object
h                     4870 non-null float32
l                     4870 non-null float32
call                  4870 non-null float32
city        

In [3]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [6]:
# number of callbacks for white-sounding names
sum(data[data.race=='w'].call)

235.0

In [8]:
# number of black-sounding names
data[data.race == 'b']['race'].count()

2435

In [15]:
# number of white-sounding names
data[data.race == 'w']['race'].count()

2435

In [10]:
# percentage of callbacks for black-sounding names
157/2435

0.06447638603696099

In [11]:
# percentage of callbacks for white-sounding names
235/2435

0.09650924024640657

1) What test is appropriate for this problem? Does CLT apply?

t-test is appropriate for this problem.
nw = nb = 2435

pb = 0.0645

pw = 0.0965

pb * nb = 0.0645 * 2435 = 157 > 5

(1-pb) * nb = (1 - 0.0645) * 2435 = 2278 > 5

So the percentage of black students been called follows normal distribution (CLT apply). 

mean_b = pb = 0.0645

std_b = sqrt(pb*(1-pb)/nb) = sqrt(0.0645*(1-0.0645)/2435) = 0.0050

Similarily, 

pw * nw = 0.0965 * 2435 = 235 > 5

(1-pw) * nw = (1 - 0.0965) * 2435 = 2200

So the percentage of white students been called follows normal distribution (CLT apply). 

mean_b = pw = 0.0965

std_b = sqrt(pw*(1-pw)/nw) = sqrt(0.0965*(1-0.0965)/2435) = 0.0060


2) What are the null and alternate hypotheses?

H_null: mean_w = mean_b
    
H_alternative: mean_w > mean_b

3) Compute margin of error, confidence interval, and p-value.

p_pool = (157 + 235)/(2435 + 2435) = 0.0805

SE_pool = sqrt(p_pool * (1 - p_pool) * (1/nb + 1/nw) = sqrt(0.0805 * (1 - 0.0805) * (1/2435 + 1/2435)) = 0.0078
               
d = pw - pb = 0.0965 - 0.0645 = 0.032
               
Set significance level alpha to be 0.05, degree of freedom = nw + nb - 1 = 4869
               
for one tail t-test, t_critical = 1.645

margin of error = t_critical * SE_pool = 1.645 * 0.0078 = 0.0128
               
confidence interval  is (- 0.0128, 0.0128)

t_score = d/SE = 0.032/0.0078 = 4.1025
               
p-value = .000021
               
d > t_critical * SE_pool or p < alpha or t > t_critical, one of the three evidences is enough to reject the null hypothesis and 
accept the alternative hypothesis.
               


4) Write a story describing the statistical significance in the context or the original problem.

Researchers are trying to examine the level of racial discrimination in the United States labor market. They randomly assigned 4870
identical résumés to black-sounding or white-sounding names and observe the number of requests for interviews from employers. Among the 2435 resumes 
assigned to black-sounding names, 157 resumes got requested for interviews from employers. Among the 2435 resumes assigned to white-sounding names, 235 resumes
got requested for interviews from employers. At a significance level of 0.05, we get a p value of 0.000021. ***p < alpha. The called ratio for white-sounding names and black-sounding names are statistically significant at level alpha = 0.05.***

5) Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

It doesn't mean that race/name is the most important factor in callback success because there is no comparsion between race/name and
other factors in the above analysis. I would use decision tree based algorithms such as random forests to train the above data and predict 
whether each resume will be called or not. The optimized random forest algorithm will output the feature importance. From there, I will
know whether the race/name is the most important factor in the callback success or not.